### PR TITLE
Add qualifier support for docker build task

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add qualifier support for docker build task
Example: https://hub.docker.com/r/opensearchstaging/opensearch/tags?page=1&name=2.0.0-alpha1
 
### Issues Resolved
Part of #1632
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/buildDockerImage.groovy [hunk 1]
```diff
@@ -23,7 +27,7 @@ void call(Map args = [:]) {
                         [
                             'bash',
                             'build-image-multi-arch.sh',
-                            "-v ${inputManifest.build.version}",
+                            "-v ${inputManifest.build.version}${build_qualifier}",
```
Comment (gaiksaya) [inline vars/buildDockerImage.groovy:30]: Its should be `-${build_qualifier}`

Comment (peterzhuamazon) [reply vars/buildDockerImage.groovy:30]: See line 9

**Review by zelinh — APPROVED** [review]
LGTM

File: tests/jenkins/jobs/BuildDockerImage_Jenkinsfile_builds_both.txt [hunk 2]
```diff
@@ -11,6 +11,6 @@
                   InputManifest.asBoolean()
                   buildDockerImage.string({name=DOCKER_BUILD_GIT_REPOSITORY, value=https://github.com/opensearch-project/opensearch-build})
                   buildDockerImage.string({name=DOCKER_BUILD_GIT_REPOSITORY_REFERENCE, value=main})
-                  buildDockerImage.string({name=DOCKER_BUILD_SCRIPT_WITH_COMMANDS, value=id && pwd && cd docker/release && curl -sSL opensearch.linux.x64 -o opensearch-x64.tgz && curl -sSL opensearch.linux.arm64 -o opensearch-arm64.tgz && bash build-image-multi-arch.sh -v 1.3.0 -f ./dockerfiles/opensearch.al2.dockerfile -p opensearch -a 'x64,arm64' -r opensearchstaging/opensearch -t 'opensearch-x64.tgz,opensearch-arm64.tgz' -n 33})
+                  buildDockerImage.string({name=DOCKER_BUILD_SCRIPT_WITH_COMMANDS, value=id && pwd && cd docker/release && curl -sSL opensearch.linux.x64 -o opensearch-x64.tgz && curl -sSL opensearch.linux.arm64 -o opensearch-arm64.tgz && bash build-image-multi-arch.sh -v 1.3.0null -f ./dockerfiles/opensearch.al2.dockerfile -p opensearch -a 'x64,arm64' -r opensearchstaging/opensearch -t 'opensearch-x64.tgz,opensearch-arm64.tgz' -n 33})
```
Comment (gaiksaya) [inline tests/jenkins/jobs/BuildDockerImage_Jenkinsfile_builds_both.txt:14]: If is null looks like `null` is getting added in the version tag

Comment (peterzhuamazon) [reply tests/jenkins/jobs/BuildDockerImage_Jenkinsfile_builds_both.txt:14]: interesting, checking

Comment (peterzhuamazon) [reply tests/jenkins/jobs/BuildDockerImage_Jenkinsfile_builds_both.txt]: fixed now.

**Comment by peterzhuamazon** [comment]
@gaiksaya @zelinh please review again I add one more test case so we can test both with/without qualifier.

File: vars/buildDockerImage.groovy [hunk 3]
```diff
@@ -3,7 +3,14 @@ void call(Map args = [:]) {
 
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: args.inputManifest))
+    def build_qualifier = inputManifest.build.qualifier
 
+    if (build_qualifier != null && build_qualifier != 'null') {
+        build_qualifier = "-" + build_qualifier
+    }
```
Comment (dblock) [inline vars/buildDockerImage.groovy:10]: Nit: move `else` up.

Comment (peterzhuamazon) [reply vars/buildDockerImage.groovy:10]: I should just default build_qualifier to null and check inputmanifest.build.qualifier.
Will change in a later improvement.

**Pull request merged**